src/sentry/api/endpoints/artifact_bundles.py

```diff
@@ -0,0 +1,110 @@
+
+        :pparam string organization_slug: the slug of the organization the
+                                          artifact bundle belongs to.
+        :pparam string project_slug: the slug of the project to list the
+                                     artifact bundles of.
+        """
+        try:
+            queryset = ProjectArtifactBundle.objects.filter(
+                organization_id=project.organization_id, project_id=project.id
+            ).values("artifact_bundle_id", "date_added")
+        except ProjectArtifactBundle.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        def expose_artifact_bundle(artifact_bundle, artifact_bundle_meta):
+            bundle_id, artifact_count = artifact_bundle_meta
+
+            return {
+                "type": "artifact_bundle",
+                "bundleId": bundle_id,
+                "date": artifact_bundle["date_added"],
+                "fileCount": artifact_count,
+            }
+
+        def serialize_results(results):
+            artifact_bundle_counts = ArtifactBundle.get_artifact_counts(
+                [r["artifact_bundle_id"] for r in results]
+            )
+
+            # We want to maintain the -date_added ordering, thus we index the metadata by using the id fetched with the
+            # first query.
+            return serialize(
+                [
+                    expose_artifact_bundle(
+                        artifact_bundle=r,
+                        artifact_bundle_meta=artifact_bundle_counts[r["artifact_bundle_id"]],
+                    )
+                    for r in results
+                    if r["artifact_bundle_id"] in artifact_bundle_counts
+                ],
+                request.user,
+            )
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by="-date_added",
+            paginator_cls=OffsetPaginator,
+            default_per_page=10,
+            on_results=serialize_results,
+        )
+
+    def delete(self, request: Request, project) -> Response:
+        """
+        Delete an Archive
+        ```````````````````````````````````````````````````
+
+        Delete all artifacts inside given archive.
+
+        :pparam string organization_slug: the slug of the organization the
+                                            archive belongs to.
+        :pparam string project_slug: the slug of the project to delete the
+                                        archive of.
+        :qparam string name: The name of the archive to delete.
+        :auth: required
+        """
+
+        bundle_id = request.GET.get("bundle_id")
+
+        if bundle_id:
+            with atomic_transaction(using=router.db_for_write(ArtifactBundle)):
+                artifact_bundle = ArtifactBundle.objects.get(
+                    organization_id=project.organization_id, bundle_id=bundle_id
+                )
+
+                if artifact_bundle is not None:
```

Comment:
AFAIK, get will raise an exception, so this `if` is redundant. 

>If get() doesn’t find any object, it raises a [Model.DoesNotExist](https://docs.djangoproject.com/en/4.1/ref/models/class/#django.db.models.Model.DoesNotExist) exception:

Reply:
That is a very good point, it was implemented like this in another endpoint and it looks like it will make it crash. Thanks for the heads up.